addons/website_sale/tests/test_website_sale_pricelist.py

```diff
@@ -522,3 +523,20 @@ def test_property_product_pricelist_multi_company(self):
         # also read a pricelist from another company if that company is the one
         # from the currently visited website.
         self.env(user=self.env.ref('base.user_demo'))['product.pricelist'].browse(demo_pl.id).name
+
+    def test_archive_pricelist_1(self):
+        ''' Test that when a pricelist is archived, the check that verify that
+            all website have at least one pricelist have access to all pricelists
+            (without considering the current allowed companies).
+        '''
+        # Make only self.c2_pl active for self.website
+        website2 = self.env.ref('website.website2')
```

Comment:
Please don't use demo data in tests.
Just
```py
self.website2 = Website.create({'name': 'Website 2'})
```

Reply:
@rdeodoo  Alright :+1: For self.website, is it normal that we browse it while it can be modified by the user?
`self.website = self.env['website'].browse(1)`

Reply:
That's kind of an exception we make in the test and always browse / use id 1 in our whole testing suite, whatever module it is.
Not ideal and probably something to rethink one day, but for now that's how it is.
It was also done by convenience, as it is not easy to create a new primary website and archive the other, you always have some kind of warning preventing that, like a SO or whatever.

Reply:
But the primary and most important rule is to not rely on demo data

Reply:
Updated. Thanks for the info!